### PR TITLE
[Review - 1] FileSystem: Init Folder and File Structures

### PR DESCRIPTION
**Pull request by satviksr**
Implement the main File System structures. 

Some APIs are very similar in file and folder. Therefore created `VbaFileSystemEntity` as the base class for `VbaFile` and `VbaFolder`. Objects of `VbaFile` and `VbaFolder` are not to be instantiated by the user. They should be instantiated only by the `FileSystem` class. 

Documentation can be found here: https://github.com/googleinterns/filesystem-in-appsscript/blob/master/docs/FileSystem.md

- [x] Created File and Folder Structures

Implemented Properties for File and Folder: 
- [x] getDateCreated
- [x] getDateLastAccessed
- [x] getFiles
- [x] name
- [x] getParentFolder
- [x] getPath
- [x] getSize
- [x] getSubFolders
- [x] getType
- [x] isRootFolder

Testing: 
- [x] File Tests
- [x] Folder Tests

### CODE REVIEW

**Review by paneendra-google — DISMISSED** [review]

File: FileSystem/VbaStructures.js [hunk 1]
```diff
@@ -156,6 +156,105 @@ function VbaDate(dateValue) {
   return this;
 }
 
+/**
+ * Adds the new entry to collection.
+ *
```
Comment (monikadhok) [inline FileSystem/VbaStructures.js]: Space?

File: FileSystem/VbaStructures.js [hunk 2]
```diff
@@ -156,6 +156,105 @@ function VbaDate(dateValue) {
...
+}
+
+/**
+ * Type to denote the end of iterator.
+ */
+var ValueIteratorEnd = function() {};
+
+/**
+ * Identifies whether the specified input is an array or array of cells (Range).
+ *
```
Comment (monikadhok) [inline FileSystem/VbaStructures.js]: Remove the space. Let us be consistent.

File: FileSystem/VbaStructures.js [hunk 3]
```diff
@@ -156,6 +156,105 @@ function VbaDate(dateValue) {
...
+ * @return {boolean} true if the given input is an array.
+ */
+function isArray(input) {
+  return Array.isArray(input) || input instanceof Array;
+}
+
+/**
+ * Helper method to verify whether an object represents end of iterator or not.
+ * @param {!Object} obj object to verify.
+ * @return {boolean} whether the object represents end of iterator or not.
```
Comment (monikadhok) [inline FileSystem/VbaStructures.js]: Whether.

**Review by monikadhok — DISMISSED** [review]
There are small small issues. Please address them before submitting.

File: FileSystem/VbaStructures.js (hunk 3 above)
Comment (satviksr) [reply FileSystem/VbaStructures.js]: Fixed

File: FileSystem/VbaStructures.js (hunk 2 above)
Comment (satviksr) [reply FileSystem/VbaStructures.js]: Fixed

File: FileSystem/VbaStructures.js (hunk 1 above)
Comment (satviksr) [reply FileSystem/VbaStructures.js]: Fixed

**Review by monikadhok — APPROVED** [review]

**Pull request merged**